### ISSUE TITLE
systemd-resolved: often returns SERVFAIL for PTR requests when DNSSEC enabled

### ISSUE THREAD

**Issue by gwire** [systemd/systemd#27361]
### systemd version the issue has been seen with

252

### Used distribution

Ubuntu 23.04

### Linux kernel version used

6.2.0-20-generic

### CPU architectures issue was seen on

x86_64

### Component

systemd-resolved

### Expected behaviour you didn't see

I expect `127.0.0.3` (which is set to use `8.8.8.8`) to return the same status as a request direct to `8.8.8.8`.

```shell-session
$ dig @8.8.8.8 +dnssec -x 54.240.25.68 | grep HEADER
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 42332
```

### Unexpected behaviour you saw

```shell-session
$ dig @127.0.0.53 +dnssec -x 54.240.25.68 | grep HEADER
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 51991
```

### Steps to reproduce the problem

* resolved is configured to use DNSSEC and `8.8.8.8` (but this applies to other DNSSEC resolvers).

```
$ resolvectl status
[...]
    Current Scopes: DNS
         Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=yes/supported
Current DNS Server: 8.8.8.8
       DNS Servers: 8.8.8.8 8.8.4.4
```

* I can usually reproduce on PTR lookups where the next label up has no NS/SOA record.

```shell-session
$ dig 145.94.167.in-addr.arpa soa +noall +answer
$ dig 1.145.94.167.in-addr.arpa ptr | grep HEADER
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 40989
$ dig @8.8.8.8 145.94.167.in-addr.arpa ptr | grep HEADER
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 46246
```

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
Apr 22 20:26:28 example systemd-resolved[69613]: [🡕] DNSSEC validation failed for question 145.94.167.in-addr.arpa IN SOA: no-signature
Apr 22 20:26:28 example systemd-resolved[69613]: [🡕] DNSSEC validation failed for question 1.145.94.167.in-addr.arpa IN DS: no-signature
Apr 22 20:26:28 example systemd-resolved[69613]: [🡕] DNSSEC validation failed for question 1.145.94.167.in-addr.arpa IN SOA: no-signature
Apr 22 20:26:28 example systemd-resolved[69613]: [🡕] DNSSEC validation failed for question 1.145.94.167.in-addr.arpa IN PTR: no-signature
Apr 22 20:26:41 example systemd-resolved[69613]: [🡕] DNSSEC validation failed for question 145.94.167.in-addr.arpa IN SOA: no-signature
```

**Comment by c---** [comment]
I don't think this is a bug. It looks like the entire in-addr.arpa domain has been broken for days. Missing DS records and such.

I can't believe more people aren't talking about this. It's a major outage. Searching doesn't yield much results.

**Comment by rpigott** [comment]
> I can usually reproduce on PTR lookups where the next label up has no NS/SOA record.

I think this is duplicated by #30856 and resolved by #31827.
